pkg/collector/listeners/docker.go

```diff
@@ -468,3 +484,18 @@ func findKubernetesInLabels(labels map[string]string) bool {
 	}
 	return false
 }
+
+// Rancher 1.x containers don't have docker networks as the orchestrator provides
+// its own CNI. The IP is stored in the `io.rancher.container.ip` label.
+func findRancherIPInLabels(labels map[string]string) (string, bool) {
+	cidr, found := labels[rancherIPLabel]
+	if found {
+		ipv4Addr, _, err := net.ParseCIDR(cidr)
+		if err != nil {
+			log.Warnf("error while retrieving Rancher IP: %q is not valid", cidr)
+		}
+		return ipv4Addr.String(), true
```

Comment:
Are you sure the `ipv4Addr` is properly set in case of `err != nil` ?
I advice something more like:
```golang
ipv4Addr, _, err := net.ParseCIDR(cidr)
if err == nil {
    return ipv4Addr.String(), true
}
log.Warnf("error while retrieving Rancher IP: %q is not valid: %s", cidr, err)
```

Reply:
@JulienBalestra addressed, can you please re-review?